### DISCUSSION TITLE
Help with WebSocket errors

### DISCUSSION THREAD

**Discussion by julio213180** [PlummersSoftwareLLC/NightDriverStrip#138] [Q&A] [answered]
Hi, I have  two Esp32 chips that i`m trying to build the spectrum analyzer, but i keep getting the same errors, can somebody help me please? i got one M5Plus stic, and one esp32 wroom32 dev kit v1, and i`m geting the same error when i try to build anything on the NightDriverStrip. i watched Daves` video on doing simple build but got the same error.

.pio\libdeps\m5plusdemo\RemoteDebug\src\utility\WebSockets.cpp:42:23: fatal error: esp32/sha.h: No such file or directory
compilation terminated.
*** [.pio\build\m5plusdemo\libe2c\RemoteDebug\utility\WebSockets.cpp.o] Error 1
.pio\libdeps\m5plusdemo\RemoteDebug\src\RemoteDebug.cpp: In member function 'void RemoteDebug::processCommand()':
.pio\libdeps\m5plusdemo\RemoteDebug\src\RemoteDebug.cpp:1520:39: warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'uint32_t {aka unsigned int}' [-Wformat=]
   DebugWS.printf("$app:M:%lu:\n", free);

Compiling .pio\build\spectrum\libeb2\RemoteDebug\utility\libb64\cdecode.c.o
.pio\libdeps\spectrum\RemoteDebug\src\utility\WebSockets.cpp:42:23: fatal error: esp32/sha.h: No such file or directory
compilation terminated.
*** [.pio\build\spectrum\libeb2\RemoteDebug\utility\WebSockets.cpp.o] Error 1

**Comment by rbergen** [comment, marked as answer]
You may need to run the following commands in the NightDriverStrip repo clone root directory:
```text
pio upgrade
pio pkg update
```

**Reply by julio213180** [reply]
I will try thank you Very much Mr. rbergen. 
apreciated.

**Comment by davepl** [comment]
That lines requires that you do a pio update to get the 4.0 framework.  If you want to stick with the old one, change it back to whatever it was before the last checkin (sorry, can’t recall from memory).

If there’s a #define that tells me what version of the expressif framework is being used, I could make that line conditional and work on both.  I’d like to avoid locking in on an old framework though.

- Dave
